cmd/upgrade/roles/cmd.go

```diff
@@ -267,15 +268,24 @@ func run(cmd *cobra.Command, argv []string) {
 
 	rolePolicyBindings, err := ocmClient.ListRolePolicyBindings(cluster.ID(), true)
 	if err != nil {
-		reporter.Errorf("Failed to get rolePolicyBinding: %s", err)
-		os.Exit(1)
+		if errors.GetType(err) == errors.PreconditionFailed {
+			reporter.Debugf("ListRolePolicyBindings is not supported in server side now")
+		} else {
+			reporter.Errorf("Failed to get rolePolicyBinding: %s", err)
+			os.Exit(1)
+		}
+	} else {
+		err = rolepolicybindings.CheckRolePolicyBindingStatus(rolePolicyBindings)
+		if err != nil {
+			reporter.Errorf("Error in rolePolicyBinding: %s", err)
+			os.Exit(1)
+		}
 	}
-	err = rolepolicybindings.CheckRolePolicyBindingStatus(rolePolicyBindings)
-	if err != nil {
-		reporter.Errorf("Error in rolePolicyBinding: %s", err)
-		os.Exit(1)
+
+	rolePolicyDetails := map[string][]aws.PolicyDetail{}
+	if rolePolicyBindings != nil {
```

Comment:
instead of checking for nil, should this be checking if the bindings are empty?
example: https://gitlab.cee.redhat.com/service/uhc-clusters-service/-/merge_requests/8086/diffs

Reply:
the nil is returned by our client: https://github.com/openshift/rosa/blob/03a121c9cb53e0daa7cd39fdf4f39e790034f923/pkg/ocm/rolepolicybinding.go#L31